invenio/modules/oauthclient/contrib/orcid.py

```diff
@@ -114,25 +114,21 @@
 
 
 def account_info(remote, resp):
-    """ Retrieve remote account information used to find local user. """
+    """Retrieve remote account information used to find local user."""
+    account_info = dict(email=None, nickname=None, orcid=resp.get("orcid"),
+                        external_id=resp.get("orcid"), external_method="orcid")
 
-    from ..handlers import token_session_key
-
-    # Store ORCID of the user in session for later retrieval
-    session[token_session_key(remote.name) + "_orcid"] = resp.get("orcid")
-    return dict(email=None, nickname=None)
+    return account_info
 
 
 def account_setup(remote, token):
-    """ Perform additional setup after user have been logged in. """
-
+    """Perform additional setup after user have been logged in."""
     from invenio.ext.sqlalchemy import db
     from ..handlers import token_session_key
 
     extra_data = {
-        "orcid": session.get(token_session_key(remote.name) + "_orcid")
+        "orcid": session.get(token_session_key(remote.name) +
+                             "_account_info").get("orcid")
```

Comment:
If you use `"_account_info").get("external_id")` then you can remove `, orcid=resp.get("orcid")` from https://github.com/inveniosoftware/invenio/pull/2600/files#diff-e35f9879d99b458ee57ff1a2343f249cR118